src/plugins/analysis/exploit_mitigations/code/exploit_mitigations.py

```diff
@@ -1,174 +1,125 @@
-        stdout=PIPE,
-        stderr=STDOUT,
+        split(f'{SHELL_SCRIPT} --file={file_path} --format=json --extended'),
+        capture_output=True,
         text=True,
-        check=False,
+        check=True,
     )
-    if checksec_process.returncode != 0:
-        raise ValueError(f'Checksec script exited with non-zero return code {checksec_process.returncode}')
-    return json.loads(checksec_process.stdout)[str(file_path)]
-
-
-def check_mitigations(file_path):
-    mitigations, summary = {}, {}
-    checksec_result = execute_checksec_script(file_path)
-
-    check_relro(file_path, mitigations, summary, checksec_result)
-    check_nx(file_path, mitigations, summary, checksec_result)
-    check_canary(file_path, mitigations, summary, checksec_result)
-    check_pie(file_path, mitigations, summary, checksec_result)
-    check_clang_cfi(file_path, mitigations, summary, checksec_result)
-    check_clang_safestack(file_path, mitigations, summary, checksec_result)
-    check_stripped_symbols(file_path, mitigations, summary, checksec_result)
-    check_runpath(file_path, mitigations, summary, checksec_result)
-    check_rpath(file_path, mitigations, summary, checksec_result)
-
-    return mitigations, summary
+    checksec_output = json.loads(checksec_process.stdout)
+    return checksec_output[file_path]
 
 
-def check_relro(file_path, mitigations, summary, checksec_result):
-    if checksec_result['relro'] == 'full':
-        summary.update({'RELRO fully enabled': file_path})
-        mitigations.update({'RELRO': 'fully enabled'})
-
-    elif checksec_result['relro'] == 'partial':
-        summary.update({'RELRO partially enabled': file_path})
-        mitigations.update({'RELRO': 'partially enabled'})
-
-    elif checksec_result['relro'] == 'no':
-        summary.update({'RELRO disabled': file_path})
-        mitigations.update({'RELRO': 'disabled'})
-
-
-def check_pie(file_path, mitigations, summary, checksec_result):
+def _check_pie(checksec_result: dict[str, str]) -> str:
     if checksec_result['pie'] == 'yes':
-        summary.update({'PIE enabled': file_path})
-        mitigations.update({'PIE': 'enabled'})
-
-    elif checksec_result['pie'] == 'no':
-        summary.update({'PIE disabled': file_path})
-        mitigations.update({'PIE': 'disabled'})
-
-    elif checksec_result['pie'] == 'dso':
-        summary.update({'PIE/DSO present': file_path})
-        mitigations.update({'PIE': 'DSO'})
-
-    elif checksec_result['pie'] == 'rel':
-        summary.update({'PIE/REL present': file_path})
-        mitigations.update({'PIE': 'REL'})
-
-    else:
-        summary.update({'PIE - invalid ELF file': file_path})
-        mitigations.update({'PIE': 'invalid ELF file'})
-
+        return 'enabled'
+    if checksec_result['pie'] == 'no':
+        return 'disabled'
+    if checksec_result['pie'] == 'dso':
+        return 'DSO'
+    if checksec_result['pie'] == 'rel':
+        return 'REL'
+    return 'invalid ELF file'
```

Comment:
On which files does this function return `'invalid ELF file'`? If this final return statement did not exist, we could ditch the function and replace it with a dictionary instead.

Reply:
Good question. Actually I have no idea, since I did not originally write it. I also couldn't reproduce it (checksec will usually fail to produce any results if the file is not a valid ELF file)

Reply:
I checked all values that are in my local DB and this is what I got:
`{'DSO': 2472, 'REL': 3874, 'disabled': 2478, 'enabled': 370, None: 494}`
Maybe it is safe to assume that this cannot happen and we can throw it out?